### PR TITLE
move conversion of _isResetOnRevisit from questionView.js to adaptModel.js

### LINKED ISSUE #2825: Framework - questionView checkIfResetOnRevisit function expects boolean
### Subject of the issue
On [L72](https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/views/questionView.js#L72) of questionView the checkIfResetOnRevisit function expects a boolean but due to this [commit](https://github.com/adaptlearning/adapt_framework/commit/b6ec1d15c6d14e7f1623f9fb5dc2632fdc023a9a) and this [PR](https://github.com/adaptlearning/adapt_framework/pull/2655) a string is returned instead. This issue causes components in Authoring Tools to enter an error state when revisiting a previously answered question.

### Your environment
* FW v5.5.1 +

### Expected behaviour
Components should restore correctly on revisit of page.

### Screenshots

Submitted, first visit.
Screenshot 2020-07-02 at 11 58 17

Still submitted, second visit. No item marking, incorrect submission CSS. 
Screenshot 2020-07-02 at 11 58 29

### PR DESCRIPTION
**Pull request by moloko**
resolves #2825 

refactor to remove nested if..else statements from `checkIfResetOnRevisit` in questionView.js

### CODE REVIEW

**Review by tomgreenfield — COMMENTED** [review]
Should this instead be handled on model instantiation?
```js
parse(data) {
  if (data._isResetOnRevisit === 'false') {
    data._isResetOnRevisit = false;
  }
  return data;
}
```
Pass `{ parse: true }` as a second parameter [here](https://github.com/adaptlearning/adapt_framework/blob/5bca8daf160fc3ce52cf0dff1307a6fd1246f447/src/core/js/data.js#L18) to trigger the function.

**Comment by moloko** [comment]
> Should this instead be handled on model instantiation?

Ooh that's a nice idea. I'll have a look at this on Monday.

File: src/core/js/data.js [hunk 1]
```diff
@@ -13,9 +13,9 @@ define([
     model(json) {
       const ModelClass = Adapt.getModelClass(json);
       if (!ModelClass) {
-        return new Backbone.Model(json);
+        return new Backbone.Model(json, { parse: true });
```
Comment (tomgreenfield) [inline src/core/js/data.js:16]: This has no effect on the default `Backbone.Model`.

**Review by tomgreenfield — APPROVED** [review]
Works for me.

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by tomgreenfield — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**